src/game/client/components/background.cpp

```diff
@@ -60,7 +60,7 @@ void CBackground::LoadBackground()
 		bool NeedImageLoading = false;
 
 		char aBuf[IO_MAX_PATH_LENGTH];
-		str_format(aBuf, sizeof(aBuf), "maps/%s", g_Config.m_ClBackgroundEntities);
+		str_format(aBuf, sizeof(aBuf), str_endswith_nocase(g_Config.m_ClBackgroundEntities, ".map") ? "maps/%s" : "maps/%s.map", g_Config.m_ClBackgroundEntities);
```

Comment:
Conditionally switching between different format strings is error prone. I prefer something like this:

```suggestion
		str_format(aBuf, sizeof(aBuf), "maps/%s%s", g_Config.m_ClBackgroundEntities, str_endswith(g_Config.m_ClBackgroundEntities, ".map") ? "" : ".map");
```

Also, use `str_endswith` for file extensions because we only support lower case file extensions to avoid any issues due different file systems.